src/Database/Attach/File.php

```diff
@@ -144,6 +145,28 @@ public function fromData($data, $filename)
         return $file;
     }
 
+    /**
+     * Creates a file object from url
+     * @param $url string URL
+     * @return $this
+     */
+    public function fromUrl($url)
+    {
+        if ($url === null) {
+            return;
+        }
+
+        $data = Http::get($url);
+
+        if ($data->code != 200) {
```

Comment:
I feel like a failed request should throw an exception because otherwise people are going to have to do something like
```php
$file->fromUrl('http://example.com/');
if (!$file) {
    // request went wrong
}
``` 
everywhere they use it